### PR TITLE
[Fixes #228] Update sample app to show more common routing patterns and add functional tests for it.

### LINKED ISSUE #228: Update sample app to show more common routing patterns and add functional tests for it
The current sample shows some obscure usage of routing. It should instead show typical patterns and extensibility. And then add functional tests to it.

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

File: test/Microsoft.AspNetCore.Routing.FunctionalTests/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+using System.Reflection;
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Routing.FunctionalTests/Properties/AssemblyInfo.cs]: copyright header

File: test/Microsoft.AspNetCore.Routing.FunctionalTests/RouteSampleWebTest.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
+using System;
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Routing.FunctionalTests/RouteSampleWebTest.cs]: copyright header

File: test/Microsoft.AspNetCore.Routing.FunctionalTests/RouteSampleWebTest.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Routing.FunctionalTests
+{
+    public class RouteSampleWebTest : IClassFixture<RoutingTestFixture<RoutingSample.Web.Startup>>
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Routing.FunctionalTests/RouteSampleWebTest.cs]: should be `RoutingSampleTest`

File: test/Microsoft.AspNetCore.Routing.FunctionalTests/async.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+namespace Microsoft.AspNetCore.Routing.FunctionalTests
+{
+    public class async
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Routing.FunctionalTests/async.cs]: ???

File: test/Microsoft.AspNetCore.Routing.FunctionalTests/RoutingTestFixture.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+using System;
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Routing.FunctionalTests/RoutingTestFixture.cs]: Copyright

File: test/Microsoft.AspNetCore.Routing.FunctionalTests/project.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    "Microsoft.AspNetCore.Routing": "1.1.0-*",
+    "Microsoft.AspNetCore.TestHost": "1.1.0-*",
+    "Microsoft.AspNetCore.Testing": "1.1.0-*",
+    "Microsoft.Extensions.DependencyInjection": "1.1.0-*",
+    "Microsoft.Extensions.Logging": "1.1.0-*",
+    "Microsoft.Extensions.Logging.Testing": "1.1.0-*",
+    "Microsoft.Extensions.WebEncoders": "1.1.0-*",
+    "Moq": "4.6.25-*",
+    "RoutingSample.Web": "1.0.0-*",
+    "xunit": "2.2.0-*"
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Routing.FunctionalTests/project.json]: Surely all of these dependencies aren't actually used here

**Comment by rynowak** [comment]
:shipit: after fixing minor issues

File: test/Microsoft.AspNetCore.Routing.FunctionalTests/Properties/AssemblyInfo.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Routing.FunctionalTests/Properties/AssemblyInfo.cs]: Can you remove these attributes? We don't need these.

File: test/Microsoft.AspNetCore.Routing.FunctionalTests/async.cs (hunk 4 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Routing.FunctionalTests/async.cs]: That's not how async works.

Comment (javiercn) [inline test/Microsoft.AspNetCore.Routing.FunctionalTests/async.cs]: I don't know how this got in here

**Pull request merged**